show-portal-user-info/src/main/java/com/esri/arcgismaps/sample/showportaluserinfo/screens/MainScreen.kt

```diff
@@ -0,0 +1,232 @@
+ */
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+private fun PortalDetails(
+    url: String,
+    onSetUrl: (String) -> Unit,
+    onSignOut: () -> Unit,
+    onLoadPortal: () -> Unit
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(8.dp),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center
+    ) {
+        // The Url text field
+        OutlinedTextField(
+            modifier = Modifier.fillMaxWidth(),
+            value = url,
+            onValueChange = onSetUrl,
+            label = { Text("Url") },
+            keyboardOptions = KeyboardOptions(
+                keyboardType = KeyboardType.Uri,
+                imeAction = ImeAction.Go
+            ),
+            keyboardActions = KeyboardActions(onAny = { onLoadPortal() }),
+            singleLine = true
+        )
+        Row(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalArrangement = Arrangement.SpaceBetween,
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            val keyboardController = LocalSoftwareKeyboardController.current
+            val focusManager = LocalFocusManager.current
+            // Clear credential button
+            Button(
+                onClick = {
+                    onSignOut()
+                    keyboardController?.hide()
+                    focusManager.clearFocus()
+                },
+                colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.tertiary)
+            ) {
+                Text(text = "Sign out")
+            }
+            // Load button
+            Button(
+                onClick = {
+                    onLoadPortal()
+                    keyboardController?.hide()
+                    focusManager.clearFocus()
+                },
+                colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.primary)
+            ) {
+                Text(text = "Load")
+            }
+        }
+    }
+}
+
+/**
+ * Displays messages to the user. This may be used to display instructions, portal info, or error messages.
+ *
+ * @param infoText the text to display
+ * @param username to display username
+ * @param email to display email
+ * @param creationDate to display creationDate
+ * @param portalName to display portalName
+ * @param userThumbnail to display userThumbnail
+ * @param isLoading whether a progress indicator should be displayed
+ * @since 200.2.0
+ */
+@Composable
+private fun InfoScreen(
+    infoText: String,
+    username: String,
+    email: String,
+    creationDate: String,
+    portalName: String,
+    userThumbnail: Bitmap?,
+    isLoading: Boolean
+) {
+    Box(
+        Modifier
+            .fillMaxSize()
+            .padding(8.dp),
+    ) {
+        LazyColumn {
+            item {
+                Text(text = infoText, modifier = Modifier.padding( bottom = 10.dp))
```

Comment:
Based on the above suggestion, the progress indicator could be added here: 
```suggestion
                Box(Modifier.fillMaxWidth()) {
                    if (isLoading) CircularProgressIndicator(
                        Modifier.align(Center).padding(10.dp)
                    )
                    else Text(
                        modifier = Modifier.align(Center).padding(10.dp),
                        textAlign = TextAlign.Center,
                        text = infoText
                    )
                }
```